Feature : Game Score 로직 구현 (2차 수정)

choitree opened the pull request:
큐 곧 일어나시겠네용 저는 이제 잡니다ㅎㅎㅎㅎㅎㅎㅎ
아침에 일어날수잇을지 모르겟네여ㅠㅠㅠㅠㅠ
구현을 해보려고 하다보니까 큐가 요청하신 내용을 추가하지 못햇어요ㅠㅠ
뭔가 조금만 더!조금만 더!! 될거같아!!이런 느낌으로 하다보니까 다섯시가 다가오네요 ㄷㄷㄷ
구현을 거의 다해갔는데 요상마상한 버그가 있네요
실행 몇번 돌려보시다보면 아시겠지만ㅋㅋㅋㅋㅋㅋㅋ
put 요청 시 tree 선수가 지혼자 다해먹어요......ㅋㅋㅋㅋㅋㅋㅋㅋ
뭔가 수정하면서 그 부분이 꼬였는걸 발견하고 헐...햇지만 이제 자야될거같아서 이만 잠에 들어볼게요ㅠㅠ
혹시 가능하시다면 리뷰 부탁드려용!~~ 그리고 큐도 이 부분 로직 구현을 한 번 시도해보시는 것도 좋을거같아요!!
진짜 복잡한데 나름 재밋더라구요!!

choitree committed:
Feature : Game Score 로직 구현 (2차 수정)

kyu-kim-kr commented:
어제 저희 이야기하기로는 이 기능 제가 아침에 하기로 했었던 것 같은데 트리가 전부 구현하셨네요? ㅎㅎ
서버에 대해서 제 위주로 맡아서 해본다는 이야기를 같이해서 서로 오해가 있었던 걸까요??
어쨋든 새벽 5시까지 고생 많으셨습니다..ㄷㄷ
코드는 pull해서 지금 동작 확인해보고 있어요. 어떻게 작성하신건지 한번 이해해보고 코멘트 달게요.

kyupid commented on BE/src/main/java/com/example/baseball/service/GameScoreService.java:
```diff
@@ -18,68 +18,92 @@
+    }
+
+
+    public PlayGameDTO findGameDTO(Integer inning, String inningStatus) {
```
이 메서드가 어떤 Game인지 찾는 메서드 맞죠??
 예를 들어서
```
Captain VS Marvel
Kyu VS Tree
...
```
이런 Game이 있다면 
이 로직이 inning과 inningStatus에 따라서 해당되는 Game을 가져오는 거죠?

kyupid commented on BE/src/main/java/com/example/baseball/service/GameScoreService.java:
```diff
@@ -18,68 +18,92 @@
-//
-//    }
+    public PlayGameDTO updatePitch(Integer inning, String inningStatus) {
+        PlayGameDTO nowGame = findGameDTO(inning, inningStatus);
+        GameScoreDTO gameScoreDTO = nowGame.getGameScoreDTO();
+        HalfInningGameDTO halfInningGameDTO = nowGame.getHalfInningGameDTO();
```
어떤 게임인지 찾는 로직이 조금 이해하기가 어렵네요 ㅠㅠ 이따가 물어봐야겠어요

kyupid commented on BE/src/main/java/com/example/baseball/service/GameScoreService.java:
```diff
@@ -18,68 +18,92 @@
+        ballCountDTOS.add(BallCountDTO.of(nowHit)); // 리스트에도 저장
+        ballCountService.updateFourBallOrThreeStrike(hitter.getName(), halfInningGameDTO.getId()); //4볼 3스트라이크 체크
+
+        if(ballCountRepository.countBallCountsByBallAndPlayerName(hitter.getName(), Sbo.HIT.toString(), halfInningGameDTO.getId()).size() == 1
+           || ballCountRepository.countBallCountsByBallAndPlayerName(hitter.getName(), Sbo.OUT.toString(), halfInningGameDTO.getId()).size() == 1) {
+            hitter.changeBattingStatus(hitter);
+            playerDTOs.remove(1);
```
타자 교체하는 로직인가요?

kyupid commented on BE/src/main/java/com/example/baseball/service/GameScoreService.java:
```diff
@@ -18,68 +18,92 @@
+
+        }
+
+        int attackPoint = ballCountService.checkScore(halfInningGameDTO.getId());
+        gameScoreDTO.updateAttackTeamScore(inningStatus, attackPoint);
+        halfInningGameDTO.updateAttackTeamScore(inningStatus, attackPoint);
```
여기도 무슨 코드인지 잘 모르겠어요 ㅎㅎ

choitree committed:
Feature : 상세페이지 확인하기 구현

- 전제 점수
- 회/이닝 별 전체 점수 리스트로
- 공격팀 선수 정보 리스트로

choitree committed:
Feature : 수비/공격 페이지 구현 (get/put)

- 전체 점수
- url에 해당하는 회/이닝의 점수
- 투수 정보
- 타자 정보
- 공격팀의 점수 목록 (strike, ball, out, hit)

choitree committed:
chore : application.properties에 url 포트번호 수정

choitree committed:
chore : 샘플데이터 추가

choitree committed:
chore : 샘플데이터 추가

kyupid committed:
fix : 대소문자 구별

choitree committed:
fix : 버그 수정

- hit 4개일 때 점수 추가
- 3아웃이 되기 전 모든 선수가 한바퀴 돌고 두 번째 차례일 때,
  이전 hit/out때문에 다음 차례 Hit/out이 안나오는 점 수정
- 처음에는 game테이블에 디비가 없고, 없을 때 추가되도록 수정
- 다음 턴 되기전에, hit/out 이외의 ball/strike 기록 지우기
-

choitree committed:
Merge branch 'tree/ballCount' of https://github.com/kyu-kim-kr/baseball into tree/ballCount

kyupid committed:
fix : 포트번호 변경

kyupid committed:
add : 선수 데이터 추가

choitree committed:
Add : /delete 기능 추가

- ballCount 데이터를 삭제할 수 있는 deleteMapping 추가
